src/flash_loader.c

```diff
@@ -47,157 +55,134 @@ static const uint8_t loader_code_stm32vl[] = {
-        0xa2, 0xf1, 0x01, 0x02, //        sub.w r2, r2, #1
-        0xef, 0xe7,             //        b.n   <next>
-        0x00, 0xbe,             // done:  bkpt  0x0000
-        0x00, 0x20, 0x02, 0x40  // flash_base:  .word 0x40022000
+        0xdf, 0xf8, 0x2c, 0xc0, 
+        0xdf, 0xf8, 0x2c, 0xa0, 
+        0xe2, 0x44, 0x03, 0x68, 
+        0x44, 0x68, 0x0b, 0x60, 
+        0x4c, 0x60, 0x00, 0xf1, 
+        0x08, 0x00, 0x01, 0xf1, 
+        0x08, 0x01, 0xba, 0xf8, 
+        0x00, 0x30, 0x13, 0xf0, 
+        0x01, 0x0f, 0xfa, 0xd0, 
+        0xa2, 0xf1, 0x01, 0x02, 
+        0x00, 0x2a, 0xee, 0xd1, 
+        0x00, 0xbe, 0x00, 0xbf, 
+        0x00, 0x20, 0x02, 0x40, 
+        0x12, 0x00, 0x00, 0x00
     };
 
 	static const uint8_t loader_code_stm32f7[] = {
         // flashloaders/stm32f7.s
-        0x08, 0x4b,
-        0x72, 0xb1,
-        0x04, 0x68,
-        0x0c, 0x60,
-        0xbf, 0xf3, 0x4f, 0x8f,        // DSB Memory barrier for in order flash write
-        0xdc, 0x89,
-        0x14, 0xf0, 0x01, 0x0f,
-        0xfb, 0xd1,
-        0x00, 0xf1, 0x04, 0x00,
-        0x01, 0xf1, 0x04, 0x01,
-        0xa2, 0xf1, 0x01, 0x02,
-        0xef, 0xe7,
-        0x00, 0xbe,                   //     bkpt	#0x00
-        0x00, 0x3c, 0x02, 0x40,
+        0xdf, 0xf8, 0x2c, 0xc0, 
+        0xdf, 0xf8, 0x2c, 0xa0, 
+        0xe2, 0x44, 0x03, 0x68, 
+        0x0b, 0x60, 0x00, 0xf1, 
+        0x04, 0x00, 0x01, 0xf1, 
+        0x04, 0x01, 0xbf, 0xf3, 
+        0x4f, 0x8f, 0xba, 0xf8, 
+        0x00, 0x30, 0x13, 0xf0, 
+        0x01, 0x0f, 0xfa, 0xd0, 
+        0xa2, 0xf1, 0x01, 0x02, 
+        0x00, 0x2a, 0xee, 0xd1, 
+        0x00, 0xbe, 0x00, 0xbf, 
+        0x00, 0x3c, 0x02, 0x40, 
+        0x0e, 0x00, 0x00, 0x00
     };
 
     static const uint8_t loader_code_stm32f7_lv[] = {
         // flashloaders/stm32f7lv.s
-        0x92, 0x00,             //      lsls    r2, r2, #2
-        0x09, 0x4b,             //      ldr r3, [pc, #36]   ; (0x20000028 <flash_base>)
-                                // next:
-        0x72, 0xb1,             //      cbz     r2, 24 <done>
-        0x04, 0x78,             //      ldrb    r4, [r0, #0]
-        0x0c, 0x70,             //      strb    r4, [r1, #0]
-        0xbf, 0xf3, 0x4f, 0x8f, //      dsb sy
-                                // wait:
-        0xdc, 0x89,             //      ldrh    r4, [r3, #14]
-        0x14, 0xf0, 0x01, 0x0f, //      tst.w   r4, #1
-        0xfb, 0xd1,             //      bne.n   e <wait>
-        0x00, 0xf1, 0x01, 0x00, //      add     r0, r0, #1
-        0x01, 0xf1, 0x01, 0x01, //      add     r1, r1, #1
-        0xa2, 0xf1, 0x01, 0x02, //      sub     r2, r2, #1
-        0xef, 0xe7,             //      b       next
-                                // done:
-        0x00, 0xbe,             //      bkpt
-        0x00, 0xbf,             //      nop
-                                // flash_base:
-        0x00, 0x3c, 0x02, 0x40  //      .word   0x40023c00
```

Comment:
We should have some comments like these for the new assembly code as well I think

Reply:
I doubt the necessity actually.
The comments should have shown people where they can find the assembly form of this flashloader binary. Maintaining another copy of the assembly is redunctant in my view.

An improvement here might be adding a note like:

**DO NOT EDIT BINARIES DIRECTLY, EDIT ASSEMBLIES INSTEAD**

Reply:
Yes, a note in the header section of this sourcecode file seems a good idea, but I would write "DO NOT MODIFY SOURCECODE DIRECTLY, EDIT ASSEMBLY FILES INSTEAD" (as binaries are commonly referred to as a compiled source code files). Correct me, shall I be wrong here.
However I just noticed that I placed my comment on the wrong file :expressionless:. This should have addressed the assembly files (.s).

Reply:
The comment in assembly files seems a bit hard to deal with. Since it is undertaken using a clean room method, I cannot add those comments to the new version myself. The code, in my view, is self-explaining enough. When I was doing the clean room research, the only odd design I encountered is about the word count (passed  in `r2`), which is not always WORD count but sometimes double word / byte count.

Reply:
I may pay attention to this later on.